liquigraph-core/src/test/java/org/liquigraph/core/configuration/ConnectionConfigurationByUriTest.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2014-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.liquigraph.core.configuration;
+
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import com.google.common.base.Supplier;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.sql.Connection;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ConnectionConfigurationByUriTest {
```

Comment:
Like in `ConnectionConfigurationByDataSourceTest`, you can cover the `catch` clause by having the mock throw a `SQLException`.